frontends/web/src/routes/buy/exchange.tsx

```diff
@@ -79,34 +76,28 @@ export const Exchange = ({ code, accounts }: TProps) => {
 
   const hasOnlyOneSupportedExchange = allExchangeDeals ? allExchangeDeals.exchanges.filter(exchange => exchange.supported).length === 1 : false;
 
-  // link locale detection to regionList to having it happen only once page load.
-  // can't use `useLoad` because `detect` function returns void.
-  useEffect(() => {
-    languageFromConfig.detect((locale: string) => setLocale(localeMainLanguage(locale)));
-  }, [regionList]);
-
   // update region Select component when `regionList` or `locale` gets populated.
   useEffect(() => {
-    if (!regionList || !locale || !config || !nativeLocale) {
+    if (!regionList || !config) {
       return;
     }
-    const regionNames = new Intl.DisplayNames([locale], { type: 'region' });
+    const regionNames = new Intl.DisplayNames([i18n.language], { type: 'region' });
     const regions = regionList.regions.map(region => ({ value: region.code, text: regionNames.of(region.code) } as TOption));
 
-    regions.sort((a, b) => a.text.localeCompare(b.text, locale));
+    regions.sort((a, b) => a.text.localeCompare(b.text, i18n.language));
     setRegions(regions);
 
     if (config.frontend.selectedExchangeRegion) {
       setSelectedRegion(config.frontend.selectedExchangeRegion);
       return;
     }
 
-    const userRegion = getRegionNameFromLocale(nativeLocale);
+    const userRegion = getRegionNameFromLocale(i18n.language);
     //Region is available in the list
     const regionAvailable = !!(regionList.regions.find(region => region.code === userRegion));
     //Pre-selecting the region
     setSelectedRegion(regionAvailable ? userRegion : '');
-  }, [regionList, locale, config, nativeLocale]);
+  }, [regionList, config]);
```

Comment:
It actually complains if I add it: 
```
src/routes/buy/exchange.tsx
  Line 100:6:  React Hook useEffect has an unnecessary dependency: 'i18n.language'. Either exclude it or remove the dependency array. Outer scope values like 'i18n.language' aren't valid dependencies because mutating them doesn't re-render the component  react-hooks/exhaustive-deps
```